go/consensus/tendermint/apps/governance/transactions.go

```diff
@@ -124,10 +125,32 @@ func (app *governanceApplication) submitProposal(
 			return err
 		}
 
-		// Ensure upgrade descriptor is far enough in future so that cancelation is still allowed.
+		// Ensure upgrade descriptor is far enough in future so that cancellation is still allowed.
 		if upgrade.Descriptor.Epoch < params.UpgradeCancelMinEpochDiff+epoch {
 			return governance.ErrUpgradeTooSoon
 		}
+
+	case proposalContent.ChangeParameters != nil:
+		// To not violate the consensus, change parameters proposals should be ignored when
+		// disabled.
+		if !params.EnableChangeParametersProposal {
+			break
```

Comment:
This is breaking, because it seems that the presence of the additional "change parameters" proposal is handled in the above `proposalContent.ValidateBasic()` and that will actually pass regardless of the value of `params.EnableChangeParametersProposal`. There needs to be a check earlier above (e.g. immediately after governance consensus parameters are fetched) to avoid problems.

Also this switch statement lacks a default branch which seems dangerous. Even though all validation is done earlier, I would add an explicit default that returns "invalid argument".